detectors/gcp/cloud-function.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	gcpFunctionNameKey = "K_SERVICE"
+)
+
+//NewCloudFunction will return an implementation for gcp cloud function resource detector
+func NewCloudFunction() resource.Detector {
+	return &CloudFunction{
+		cloudRun: NewCloudRun(),
+	}
+}
+
+type CloudFunction struct {
+	cloudRun *CloudRun
+}
```

Comment:
The other detectors in this package all provide a type without a constructor function. Can we match the existing initialization method for the package and have `CloudFunction.Detect` do any setup it needs to instead of maintain a `*CloudRun` referenct?

If not, `CloudFunction` needs to be unexported so users do not create an invalid value that cannot be used.

Reply:
Hi @MrAlias ,

True, that it is quite different from implementations like that of, GCE and GKE.
Current implementation of a constructor function, returning a referenct, is inspired from,
CloudRun implementation from same package.

Also, have kept it exporter, because same is done for CloudRun implementation.

Can keep it private, if required.
Please let me know on this.